custom_model_runner/datarobot_drum/drum/drum.py

```diff
@@ -334,15 +339,27 @@ def _prepare_prediction_server_or_batch_pipeline(self, run_language):
                 {
                     "host": host,
                     "port": port,
-                    "threaded": str(options.threaded).lower(),
                     "show_perf": str(options.show_perf).lower(),
+                    "engine_type": "RestModelServing" if options.production else "Generic",
+                    "component_type": "uwsgi_serving"
+                    if options.production
+                    else "prediction_server",
                 }
             )
 
         functional_pipeline_str = CMRunnerUtils.render_file(
             functional_pipeline_filepath, replace_data
         )
 
+        if self.run_mode == RunMode.SERVER:
+            if options.production:
+                pipeline_json = json.loads(functional_pipeline_str)
+                # mlpiper requires model file to be provided in a specific param.
+                # We just provide current file, but never use.
+                if json_fields.PIPELINE_SYSTEM_CONFIG_FIELD not in pipeline_json:
+                    system_config = {"modelFileSourcePath": os.path.abspath(__file__)}
+                pipeline_json[json_fields.PIPELINE_SYSTEM_CONFIG_FIELD] = system_config
+                functional_pipeline_str = json.dumps(pipeline_json)
```

Comment:
I don't get what this all does.

Reply:
There is a comment:
```
mlpiper requires model file to be provided in a specific param.
We just provide current file, but never use.
```

Reply:
Let me try:

"Because of tech debt in MLPiper which requires that the modelFileSourcePath key be filled with something, we're putting in a dummy file path here"

Reply:
Or you could go fix MLPiper

Reply:
fixed comment

Reply:
**DONE**